fili-core/src/test/groovy/com/yahoo/bard/webservice/util/IntervalUtilsSpec.groovy

```diff
@@ -326,6 +328,32 @@ class IntervalUtilsSpec extends Specification {
         DAY   | ["2012-02-03/2017-04-04"] | ["2012-02-03/2012-05-04", "2017-02-03/2017-04-04"]
     }
 
+    @Unroll
+    def "getCoarsestTimeGrain returns duration of #expected among #duration1, #duration2, #duration3"() {
```

Comment:
Yes. It's better to keep the tests focused on the external interface and the behavior, and to try to not duplicate what the code under test is doing when testing it.